test/error_messages.coffee

```diff
@@ -866,7 +867,7 @@ test "unexpected object keys", ->
     {a: 1, [[]]: 2}
            ^
   '''
-
+###
```

Comment:
Believe it or not, `{a: 1, [[]]: 2}` is valid JavaScript. Go ahead, try it!

So we should rewrite this test to check as appropriate and move it out of the `error_messages` file. And make `[[]]:` compilable if it isn’t already.